controllers/capabilities/openapi_product_reconciler.go

```diff
@@ -369,3 +377,70 @@ func (p *OpenAPIProductReconciler) desiredPrivateAPISecurity() *capabilitiesv1be
 
 	return privateAPISec
 }
+
+func (p *OpenAPIProductReconciler) desiredOIDCSecurity() *capabilitiesv1beta1.SecuritySpec {
+	if p.openapiCR.Spec.SecurityScheme == nil && p.openapiCR.Spec.SecurityScheme.Type == nil {
+		return nil
+	}
+
+	privateAPISec := &capabilitiesv1beta1.SecuritySpec{}
+
+	if *p.openapiCR.Spec.SecurityScheme.Type == "openIdConnect" &&
+		p.openapiCR.Spec.SecurityScheme.OpenIdConnectUrl != nil {
+		privateAPISec.OpenApiSecuritySchemeType = p.openapiCR.Spec.SecurityScheme.Type
+		privateAPISec.OpenIdConnectUrl = p.openapiCR.Spec.SecurityScheme.OpenIdConnectUrl
+	}
+
+	return privateAPISec
+}
+
+func (p *OpenAPIProductReconciler) desiredOIDCAuthentication(secReq *helper.ExtendedSecurityRequirement) *capabilitiesv1beta1.AuthenticationSpec {
+	authSpec := &capabilitiesv1beta1.AuthenticationSpec{
+		OIDC: &capabilitiesv1beta1.OIDCSpec{
+			IssuerType: "keycloak",
+			Security:   p.desiredOIDCSecurity(),
+			AuthenticationFlow: &capabilitiesv1beta1.OIDCAuthenticationFlowSpec{
+				StandardFlowEnabled:       false,
+				ImplicitFlowEnabled:       false,
+				DirectAccessGrantsEnabled: false,
+				ServiceAccountsEnabled:    false,
+			},
+		},
+	}
+
+	p.setOIDCAuthenticationParams(authSpec, secReq)
+
+	return authSpec
+}
+
+func (p *OpenAPIProductReconciler) setOIDCAuthenticationParams(authSpec *capabilitiesv1beta1.AuthenticationSpec, secReq *helper.ExtendedSecurityRequirement) {
```

Comment:
> The flow is determined by the OpenAPI document instead of the OpenAPI CR.

Thank you for comment. But not so clear, as you also "define" (bool) flows in CR in https://github.com/3scale/3scale-operator/issues/842

Reply:
discussed